### PR TITLE
devcontainer: add devcontainer for debugging linux builds

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Adding devocontainer to debug linux builds for formulae like [oil 0.12.9](https://github.com/Homebrew/homebrew-core/pull/116717) and bazel build.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "homebrew/ubuntu22.04",
+  "image": "ghcr.io/homebrew/ubuntu22.04:latest",
```
Comment (MikeMcQuaid) [inline .devcontainer/devcontainer.json:3]:
```suggestion
  "image": "ghcr.io/homebrew/brew:latest",
```

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  "workspaceFolder": "/home/linuxbrew/.linuxbrew/Homebrew",
```
Comment (MikeMcQuaid) [inline .devcontainer/devcontainer.json:4]:
```suggestion
  "workspaceFolder": "/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core",
```

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  "workspaceMount": "source=${localWorkspaceFolder},target=/home/linuxbrew/.linuxbrew/Homebrew,type=bind,consistency=cached",
```
Comment (MikeMcQuaid) [inline .devcontainer/devcontainer.json:5]:
```suggestion
  "workspaceMount": "source=${localWorkspaceFolder},target=/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core,type=bind,consistency=cached",
```

**Review by MikeMcQuaid — COMMENTED** [review]
Would be good to also have a `.devcontainer/on-create-command.sh` that ensures that e.g. `brew audit`, `brew style`, etc. can have dependencies prebuilt.

**Comment by chenrui333** [comment]
> Would be good to also have a `.devcontainer/on-create-command.sh` that ensures that e.g. `brew audit`, `brew style`, etc. can have dependencies prebuilt.

make sense, I will add that later.

**Comment by chenrui333** [comment]
I actually found a easy workaround to develop/debug formula, it is documented in this workflow, https://github.com/chenrui333/github-action-test/blob/main/.github/workflows/brew.yml

```yml
  # debug formula on macos
  formula-debug-macos:
    runs-on: macos-12
    if: github.event_name == 'pull_request'
    steps:
      - name: Setup tmate session
        uses: mxschmitt/action-tmate@v3

  # debug formula on ubuntu
  formula-debug-ubuntu:
    runs-on: ubuntu-22.04
    if: github.event_name == 'pull_request'
    steps:
      # As brew got removed from the ubuntu-latest image, we need to install it
      # https://github.com/actions/runner-images/issues/6283
      - name: Set up Homebrew
        id: set-up-homebrew
        uses: Homebrew/actions/setup-homebrew@449449e64aaa5faac0d24229a2ffba32438e4b63
      - name: Setup tmate session
        uses: mxschmitt/action-tmate@v3
```

**Pull request closed without merging**